executor/explain_test.go

```diff
@@ -424,3 +424,48 @@ func TestFix29401(t *testing.T) {
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;`)
 	tk.MustExec(" explain select /*+ inl_hash_join(t1) */ * from tt123 t1 join tt123 t2 on t1.b=t2.e;")
 }
+
+func TestFix35149(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t9;")
+	tk.MustExec(`CREATE TABLE t9 (
+		id varchar(100) NOT NULL,
+		id2 int(11) NOT NULL,
+		PRIMARY KEY (id,id2) /*T![clustered_index] CLUSTERED */
+	  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;`)
+	tk.MustExec("drop table if exists t10;")
+	tk.MustExec(`CREATE TABLE t10 (
+		id varchar(100) NOT NULL,
+		id2 int(11) DEFAULT NULL,
+		PRIMARY KEY (id) /*T![clustered_index] CLUSTERED */
+	  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin`)
+	tk.MustExec(`CREATE TABLE t11 (
+		id varchar(100) NOT NULL ,
+		id2 int(11) DEFAULT NULL,
+		PRIMARY KEY (id) /*T![clustered_index] CLUSTERED */
+	  ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin`)
+	tk.MustExec("set names utf8mb4 collate utf8mb4_general_ci;")
+	tk.MustQuery("explain select /*+ INL_JOIN(t9,t10) */ * from t9 join t10 on t9.id = t10.id and  t10.id = 's12' ;").Check(testkit.RowsWithSep("|",
+		`HashJoin_7|100.00|root| CARTESIAN inner join`, `├─Point_Get_11(Build)|1.00|root|table:t10, clustered index:PRIMARY(id) `, `└─TableReader_10(Probe)|10.00|root| data:TableRangeScan_9`,
+		`  └─TableRangeScan_9|10.00|cop[tikv]|table:t9|range:["s12","s12"], keep order:false, stats:pseudo`))
```

Comment:
When create table collation is `_bin`, and the session collate is `_general_ci`, `INL_JOIN` join hint can't work?

Reply:
In scenarios where pushdown is possible, hash join will be used